esmvaltool/diag_scripts/shared/_validation.py

```diff
@@ -84,20 +93,37 @@ def get_control_exper_obs(short_name, input_data, cfg, cmip_type):
 
     # determine CONTROL and EXPERIMENT datasets
 
-    # corner case: they could be the same dataset name
-    if cfg['control_model'] == cfg['exper_model']:
-        logger.info("Identical Control/Experiment dataset names: %s",
-                    dataset_selection[0]['dataset'])
-        control, experiment = _disentagle_iden_datasets(dataset_selection)
-        return control, experiment, obs_selection
+    # #nocornercase
+    # corner case not necessary when using distinct dataset aliases
+    # # corner case: they could be the same dataset name
+    # if cfg['control_model'] == cfg['exper_model']:
+    #     logger.info("Identical Control/Experiment dataset names: %s",
+    #                 dataset_selection[0]['dataset'])
+    #     control, experiment = _disentagle_iden_datasets(dataset_selection)
+    #     return control, experiment, obs_selection
 
     # if they're not the same dataset, fire away
+
+    # make sure the chosen datasets for control and exper are available
+    alias_selection = [model['alias'] for model in dataset_selection]
+
+    if cfg['control_model'] not in alias_selection:
+        logger.error("Control dataset %s not in datasets",
+                     cfg['control_model'])
+        sys.exit(1)
```

Comment:
it's better to raise an error than to `sys.exit()` - so here, you should do `raise ValueError(f"Control dataset {cfg['control_model']} not in datasets")`

Reply:
Great Idea! Done!